src/containers/app/CreateApp.js

```diff
@@ -0,0 +1,170 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Immutable from 'immutable';
+
+import { FormControl, FormGroup, ControlLabel, Button, Alert } from 'react-bootstrap';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+
+import AsyncContent from '../../components/asynccontent/AsyncContent';
+import { createAppRequest } from './AppActionFactory';
+
+
+class CreateApp extends React.Component {
+
+  static propTypes = {
+    actions: PropTypes.shape({
+      createAppRequest: PropTypes.func.isRequired
+    }).isRequired,
+    createAppAsyncState: PropTypes.instanceOf(Immutable.Map).isRequired
+  }
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      title: '',
+      description: '',
+      name: '',
+      appTypeIds: [],
+      url: '',
+      isError: false
+    };
+  }
+
+  onTitleChange = (event) => {
+    this.setState({
+      title: event.target.value
+    });
+  };
+
+  onNameChange = (event) => {
+    this.setState({
+      name: event.target.value
+    });
+  };
+
+  onDescriptionChange = (event) => {
+    this.setState({
+      description: event.target.value
+    });
+  };
+
+  onUrlChange = (event) => {
+    this.setState({
+      url: event.target.value
+    });
+  };
+
+  onAppTypeIdsChange = (event) => {
+    this.setState({
+      appTypeIds: event.target.value
+    });
+  };
+
+  onSubmit = () => {
+    // TODO: Utilize Models and Model Builders
+    const { title, description, name, appTypeIds, url } = this.state;
+    // need to separate the appTypeIds string by comma, put in array
+    if (!title || !description || !name || !appTypeIds || !url) {
+      // Alert! Prevent form submission
+      this.setState({
+        isError: true
+      });
+      return;
+    }
+    const splitAppTypeIds = appTypeIds
+      .split(',')
+      .map(id => id.trim())
+      .filter(id => !!id);
+    const app = {
```

Comment:
The style we follow is to put all shorthand properties first, followed by standard properties, and alphabetize them:

```
const app = {
  description,
  name,
  title,
  url,
  appTypeIds: splitAppTypeIds
}
```